### PR TITLE
Log in with OIDC not work with MINIO_DOMAIN (#8558)

### LINKED ISSUE #8558: Log in with OIDC not work with MINIO_DOMAIN
Can't log in to MinIO Browser with OIDC when defined `MINIO_DOMAIN` environment variable.

## Expected Behavior

Successful login.

## Current Behavior

Log in failed.

## Steps to Reproduce (for bugs)
1. Run etcd ([MinIO docs](https://github.com/minio/minio/blob/master/docs/sts/etcd.md))
2. Run MinIO with `MINIO_DOMAIN`
```
docker run --rm --name minio -p 9000:9000  \
  -e "MINIO_ACCESS_KEY=minio" \
  -e "MINIO_SECRET_KEY=minio123" \
  -e "MINIO_IDENTITY_OPENID_STATE=on" \
  -e "MINIO_IDENTITY_OPENID_CONFIG_URL=https://sso.company.com/auth/realms/realm/.well-known/openid-configuration" \
  -e "MINIO_ETCD_ENDPOINTS=http://host.docker.internal:2379" \
  -e "MINIO_DOMAIN=s3.company.com" \
  minio/minio:edge server /data
```
3. Try log in with OpenID

### PR DESCRIPTION
**Pull request by ihostage**
## Description

Fixes #8558 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

**Review by harshavardhana — APPROVED** [review]

File: cmd/generic-handlers.go [hunk 1]
```diff
@@ -279,6 +280,16 @@ func isAdminReq(r *http.Request) bool {
 	return strings.HasPrefix(r.URL.Path, adminPathPrefix)
 }
 
+// guessIsLoginSTSReq - returns true if incoming request is Login STS user
+func guessIsLoginSTSReq(req *http.Request) bool {
+	if req == nil {
+		return false
+	}
+	return strings.HasPrefix(req.URL.Path, loginPathPrefix) ||
+		(req.Method == http.MethodPost && req.URL.Path == "/" &&
```
Comment (harshavardhana) [inline cmd/generic-handlers.go]: use SlashSeparator const instead of "/"

File: cmd/globals.go [hunk 2]
```diff
@@ -265,6 +265,7 @@ func getGlobalInfo() (globalInfo map[string]interface{}) {
 	globalInfo = map[string]interface{}{
 		"isWorm":       globalWORMEnabled,
 		"serverRegion": globalServerRegion,
+		"domains":      globalDomainNames,
```
Comment (harshavardhana) [inline cmd/globals.go]: We may remove this globalInfo entirely, let's not extend its scope for now.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :x: [more...](#83fbcec9-00cd-4b1e-bbc0-c0b1bf0d913d) |
| mint-large-bucket.sh | :x: [more...](#d15f5825-cc93-4b38-8663-94cc468596a1) |
<a name="83fbcec9-00cd-4b1e-bbc0-c0b1bf0d913d"></a>
### 8559-d721494/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.56:32569
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 2c21e22e605c:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 5 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-r4gy4xoblozfm4",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 364964,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 15D9511E5ADDB62A, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="d15f5825-cc93-4b38-8663-94cc468596a1"></a>
### 8559-d721494/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.56:30321
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp c50b0fdcdbb6:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 55 seconds
(4/14) Running aws-sdk-ruby tests ... done in 5 seconds
(5/14) Running awscli tests ... done in 1 minutes and 22 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 22 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 36 seconds
(9/14) Running minio-go tests ... FAILED in 3 minutes and 48 seconds
{
  "args": {},
  "duration": 80975,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
Error: No such image: minio/minio:8559-d721494

File: cmd/generic-handlers.go (hunk 1 above)
Comment (ihostage) [reply cmd/generic-handlers.go]: 👌

File: cmd/globals.go (hunk 2 above)
Comment (ihostage) [reply cmd/globals.go]: 👌

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :x: [more...](#1c24368d-1c6f-4a9d-a501-7403f3a4c99b) |
| mint-large-bucket.sh | :x: [more...](#3ead1dee-052f-4e39-bf7c-bb113195210c) |
<a name="1c24368d-1c6f-4a9d-a501-7403f3a4c99b"></a>
### 8559-dbe2fe4/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.61:31064
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 1a2e20ed0567:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 4 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-qjs00qtu0wg3ji",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 364850,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 15D952ABA18E0674, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="3ead1dee-052f-4e39-bf7c-bb113195210c"></a>
### 8559-dbe2fe4/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:31736
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp bb1a4a2ca42c:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 2 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 51 seconds
(4/14) Running aws-sdk-ruby tests ... done in 5 seconds
(5/14) Running awscli tests ... done in 2 minutes and 9 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... FAILED in 21 seconds
{
  "name": "mc",
  "duration": "8457",
  "function": "test_get_object_multipart",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-13474 --quiet --no-color cp myminio/mc-test-bucket-21466/mc-test-object-5199 mc-test-object-5199.downloaded  >>> \n`myminio/mc-test-bucket-21466/mc-test-object-5199` -> `mc-test-object-5199.downloaded`\nmc: <ERROR> Failed to copy `http://72.28.97.55:31736/mc-test-bucket-21466/mc-test-object-5199`. read tcp 172.17.0.9:56256->72.28.97.55:31736: read: connection reset by peer\nmc: <ERROR> Session safely terminated. To resume session `mc session resume gqSjVeKz`"
}

Executed 6 out of 14 tests successfully.
```
Error: No such image: minio/minio:8559-dbe2fe4

**Review by harshavardhana — APPROVED** [review]

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#65fd427d-8a98-4336-98b3-385f1129246a) |
<a name="65fd427d-8a98-4336-98b3-385f1129246a"></a>
### 8559-53ee492/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.61:32650
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 2a2942e3a677:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 2 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 53 seconds
(4/14) Running aws-sdk-ruby tests ... done in 7 seconds
(5/14) Running awscli tests ... done in 2 minutes and 18 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 21 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 12 seconds
(9/14) Running minio-go tests ... FAILED in 3 minutes and 34 seconds
{
  "args": {},
  "duration": 78449,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
Error: No such image: minio/minio:8559-53ee492

**Pull request merged**